adapters/tappx/tappx_test.go

```diff
@@ -2,10 +2,31 @@ package tappx
 
 import (
 	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/stretchr/testify/assert"
 	"net/http"
+	"regexp"
 	"testing"
 )
 
 func TestJsonSamples(t *testing.T) {
 	adapterstest.RunJSONBidderTest(t, "tappxtest", NewTappxBidder(new(http.Client), "https://{{.Host}}"))
 }
+
```

Comment:
Would be good to add some more test cases for the `buildEndpointURL` method like passing an invalid `host` which will lead to `url.Parse(host)` throwing an error, passing empty `tappxkey`, etc

Tip: You can look at table driven test cases to make it easier to add new test cases and reduce redundant code with multiple test cases. Ref: https://github.com/golang/go/wiki/TableDrivenTests

Reply:
added test case for url.Parse(host) throwing error.
empty parameter tests should be already covered through JSONBidder tests
there are still some errors that aren't covered like resolving macros or JSON parsing errors, if those are necessary please send us a full list and we will try to write them all

Reply:
Sounds good. Yeah, that should cover most cases. 

The only reason behind that suggestion was that `JSONBidderTests` are more testing the `MakeRequest` and `MakeBids` function and even though `buildEndpointURL` is used as part of the `MakeRequest` function,  breaking up the unit tests by each function helps to get better coverage.  For example, if we have good unit tests written for `buildEndpointURL` and all the other functions that `MakeRequest` and `MakeBids` call then we don't have to worry about the different places those functions can fail when writing unit tests for `MakeRequest` and `MakeBids` themselves. It just makes the tests more readable and manageable.

I understand that doing that in this PR can make it a little bloated but just something to consider for next time :)